src/smtp-lib-test/raw_socket_test.cpp

```diff
@@ -0,0 +1,116 @@
+#include "catch/catch.hpp"
+
+#include "smtp-lib/connection.hpp"
+
+#include "smtp-lib-test/helpers.hpp"
+
+TEST_CASE("open socket", "[raw_socket]") {
+    auto socket = RawSocket::new_socket();
+
+    CHECK(socket.is_valid());
+}
+
+TEST_CASE("bind socket", "[raw_socket]") {
+    auto socket = RawSocket::new_socket();
+    uint16_t port = 5555;
```

Comment:
alle these tests will fail, if port 5555 is blocked. can we set this to 0, let the OS decide on a free port and then do something like `socket.get_port_number()`?

Reply:
Its related to the initial problems of freeing the resources correctly. We could do it but i think we would miss out port leaks then.